src/aat/java/uk/gov/hmcts/ccd/fta/steps/BackEndFunctionalTestScenarioPlayer.java

```diff
@@ -0,0 +1,259 @@
+package uk.gov.hmcts.ccd.fta.steps;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import feign.FeignException;
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import io.restassured.specification.QueryableRequestSpecification;
+import io.restassured.specification.RequestSpecification;
+import io.restassured.specification.SpecificationQuerier;
+import org.junit.Assert;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.cucumber.core.api.Scenario;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import uk.gov.hmcts.ccd.datastore.tests.AATHelper;
+import uk.gov.hmcts.ccd.datastore.tests.helper.idam.AuthenticatedUser;
+import uk.gov.hmcts.ccd.fta.data.RequestData;
+import uk.gov.hmcts.ccd.fta.data.ResponseData;
+import uk.gov.hmcts.ccd.fta.data.UserData;
+import uk.gov.hmcts.ccd.fta.util.JsonUtils;
+
+public class BackEndFunctionalTestScenarioPlayer implements BackEndFunctionalTestAutomationDSL {
+
+    private static final String DYNAMIC_CONTENT_PLACEHOLDER = "[[DYNAMIC]]";
+
+    private final BackEndFunctionalTestScenarioContext scenarioContext;
+    private final AATHelper aat;
+    private Scenario scenario;
+
+    private Logger logger = LoggerFactory.getLogger(BackEndFunctionalTestScenarioPlayer.class);
+
+    public BackEndFunctionalTestScenarioPlayer() {
+        aat = AATHelper.INSTANCE;
+        RestAssured.baseURI = aat.getTestUrl();
+        RestAssured.useRelaxedHTTPSValidation();
+        scenarioContext = new BackEndFunctionalTestScenarioContext();
+    }
+
+    @Before()
+    public void prepare(Scenario scenario) {
+        this.scenario = scenario;
+    }
+
+    @Override
+    @Given("an appropriate test context as detailed in the test data source")
```

Comment:
Since this step will AFAIK be used as the first step of ALL scenarios could be an idea to move this code implementation into the @before step it self and remove this Step from the feature files as readying it brings little value to the reader and removing it would help condense the Scenarios - probably best to discuss with team first before implementing his change

Reply:
This branched contained that code temporarily and the review comment will be addressed in the actual branch editing this class.